### PR TITLE
Update SQL_Injection_Prevention_Cheat_Sheet.md

### PR DESCRIPTION
**Pull request by jmanico**
Major cleanup

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as [TEXT](URL)
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

**Comment by jmanico** [comment]
Thank you to Robert Thornton for this work!

File: cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.md [hunk 1]
```diff
@@ -422,7 +252,7 @@ In addition to being a primary defense when nothing else is possible (e.g., when
 
 **SQL Injection Attack Cheat Sheets**:
 
-The following articles describe how to exploit different kinds of SQL Injection Vulnerabilities on various platforms that this article was created to help you avoid:
+The following articles describe how to exploit different kinds of SQL injection vulnerabilities on various platforms (that this article was created to help you avoid):
```
Comment (rsrinivasanhome) [inline cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.md:255]: that this article was created to help you avoid SQL Injection

File: cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.md [hunk 2]
```diff
@@ -2,32 +2,23 @@
...
 
-SQL Injection flaws are introduced when software developers create dynamic database queries constructed with string concatenation which includes user supplied input. To avoid SQL injection flaws is simple. Developers need to either: a) stop writing dynamic queries with string concatenation; and/or b) prevent user supplied input which contains malicious SQL from affecting the logic of the executed query.
+## What Is a SQL Injection Attack?
 
-This article provides a set of simple techniques for preventing SQL Injection vulnerabilities by avoiding these two problems. These techniques can be used with practically any kind of programming language with any type of database. There are other types of databases, like XML databases, which can have similar problems (e.g., XPath and XQuery injection) and these techniques can be used to protect them as well.
+Attackers can use SQL injection on an application if it has dynamic database queries that use string concatenation and user supplied input. To avoid SQL injection flaws, developers need to:
 
-**Primary Defenses:**
```
[the remark this answers is not in the document]
Comment (jmanico) [reply cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.md:14]: The example in 1201 points to a dynamic query in a stored procedure if I am looking at the right code.  I cant accept this PR as is, its a big change to a CS that we just made big updates on. Can you perhaps drop this PR and maybe give me several small PR's?

File: cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.md [hunk 3]
```diff
@@ -36,28 +27,27 @@ try {
...
-Language specific recommendations:
+When developers are taught how to write database queries, they should be told to use prepared statements with variable binding (aka parameterized queries). Prepared statements are simple to write and easier to understand than dynamic queries and parameterized queries force the developer to define all SQL code first and pass in each parameter to the query later.
 
-- Java EE – use `PreparedStatement()` with bind variables
-- .NET – use parameterized queries like `SqlCommand()` or `OleDbCommand()` with bind variables
-- PHP – use PDO with strongly typed parameterized queries (using bindParam())
-- Hibernate - use `createQuery()` with bind variables (called named parameters in Hibernate)
-- SQLite - use `sqlite3_prepare()` to create a [statement object](http://www.sqlite.org/c3ref/stmt.html)
+If database queries use this coding style, the database will always distinguish between code and data, regardless of what user input is supplied.
+Also, prepared statements ensure that an attacker is not able to change the intent of a query, even if SQL commands are inserted by an attacker.
```
[the remark this answers is not in the document]
Comment (jmanico) [reply cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.md:46]: Parameterized queries **do not escape under the hood**, its about the ordering of query plan construction.

**Pull request merged**